### PR TITLE
Add ability to delete previously stored images

### PR DESCRIPTION
**Pull request by glennhartmann**
Live at https://llaminator-dev.netlify.app/

Just making a note of some weird unexpected behaviour I noticed:

The current PR sends an event up the DOM to let `llaminator.ts` handle things. Before that, I tried letting the `llama-item` remove itself from the DOM (`this.parentElement.remove(this)`, or something like that).

This worked as expected in that moment, but then upon uploading a new image, the UI wouldn't update (despite `render()` being called with the proper elements in response to an upload).

I did confirm that the image was uploaded and the db instance was up-to-date, and that `render()` was in fact getting called, but the newly-uploaded images wouldn't be visible until I refreshed the page.

Even more confusingly, everything did actually update properly if I stepped through the code with devtools....

If I uploaded a new image normally, then used devtools when uploading _another_ new image, then only the image uploaded while devtools was running ended up being displayed (until page refresh).

No idea what's going on there - I'm sure it's something about `lit` trying to be clever about which elements changed and what exactly needs to be re-rendered, but wow is that prone to subtle unexpected undebuggable bugs.

### CODE REVIEW

File: src/components/llama-item.ts [hunk 1]
```diff
@@ -75,7 +96,12 @@ export class LlamaItem extends LitElement {
                       @click=${this.shareItem}>
                 share
               </button>`}
-            <!-- TODO: Add the option to remove this item -->
+            <button class="material-icons mdc-icon-button mdc-card__action mdc-card__action--icon"
+                    title="Share this item"
```
Comment (glennhartmann) [inline src/components/llama-item.ts:100]: N2S: fix this text

File: src/components/llama-item.ts [hunk 2]
```diff
@@ -56,6 +56,27 @@ export class LlamaItem extends LitElement {
     }
   }
 
+  /**
+   * Deletes an item from storage and removes it from the UI.
+   *
+   * @todo Should the db interaction be happening here or in llaminator.ts? Same with
+   *     shareItem() - should it be an event that llaminator.ts handles?
```
Comment (beverloo) [inline src/components/llama-item.ts:63]: High-level thoughts on this: we aren't currently following any particular architecture paradigms, but I think that moving towards MVVM would align well with what we need.

**Review by beverloo — APPROVED** [review]
Thank you, also for the demo page!

There is one new issue that I'm observing (but we can address separately): add `foo.jpg`, delete it, then add `foo.jpg` again. It refuses to add the image, with no warnings/messages/anything. Interestingly, adding `bar.jpg` after the delete works fine.

**Comment by glennhartmann** [comment]
> Thank you, also for the demo page!
> 
> There is one new issue that I'm observing (but we can address separately): add `foo.jpg`, delete it, then add `foo.jpg` again. It refuses to add the image, with no warnings/messages/anything. Interestingly, adding `bar.jpg` after the delete works fine.

That sounds like a very odd issue given that re-adding the same image will generate a new random ID - that entry in the db will be completely separate from the previous one, and the `id` on the DOM element will also be different. I can't think of a reason Llaminator would make any association between the recently-deleted item and the new one, even if they have the same image data and file name.

**Pull request merged**